frame/staking/src/lib.rs

```diff
@@ -927,8 +1094,82 @@ decl_module! {
 
 		fn deposit_event() = default;
 
-		fn on_initialize() {
+		/// Does the following:
+		///
+		/// 1. potential storage migration
+		/// 2. sets `ElectionStatus` to `Open(now)` where `now` is the block number at which
+		///    the election window has opened. The offchain worker, if applicable, will execute at
+		///    the end of the current block. `submit_election_solution` will accept solutions from
+		///    this block until the end of the era.
+		fn on_initialize(now: T::BlockNumber) {
 			Self::ensure_storage_upgraded();
+			// @guillaume Is using active era okay here? not current era? These two tests are related
+			// offchain_election_flag_is_triggered and is_current_session_final_works
+			let maybe_current_era = Self::active_era();
+			if maybe_current_era.is_none() { return; }
+			let current_era = maybe_current_era.expect("value checked to be 'Some'").index;
```

Comment:
I don't understand why you are checking active_era is some here, the code below is unrelated to active_era to me.
```suggestion
```

Reply:
Yea the code is unrelated and I left it there so that it generates warnings. What I would like to double check is the two test cases I named.